src/org/exist/collections/Collection.java

```diff
@@ -804,28 +816,6 @@ public DocumentImpl getDocumentWithLock(final DBBroker broker, final XmldbURI ur
         }
     }
 
-    public DocumentImpl getDocumentNoLock(final DBBroker broker, final String rawPath) throws PermissionDeniedException {
```

Comment:
it should be `Deprecated` one release as minimum. Let's follow some rules, not just removing methods because they looks wrong.